File: Web/src/services/trainingService.ts
import { apiClient } from '../core/network/apiClient';
import type { TrainingRecordCreateRequest, TrainingRecordResponse, TrainingStatisticsResponse } from '../models/trainingModels';

export const trainingService = {
  async create(body: TrainingRecordCreateRequest): Promise<TrainingRecordResponse> {
    const { data } = await apiClient.post<TrainingRecordResponse>('/api/v1/trainings', body);
    return data;
  },
  async list(): Promise<TrainingRecordResponse[]> {
    const { data } = await apiClient.get<TrainingRecordResponse[]>('/api/v1/trainings');
    return data;
  },
  async statistics(): Promise<TrainingStatisticsResponse> {
    const { data } = await apiClient.get<TrainingStatisticsResponse>('/api/v1/trainings/statistics');
    return data;
  },
};

File: Android/app/src/main/java/com/diveapp/android/service/TrainingService.kt
package com.diveapp.android.service

import com.diveapp.android.model.TrainingRecordCreateRequest
import com.diveapp.android.model.TrainingRecordResponse
import com.diveapp.android.model.TrainingStatisticsResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface TrainingService {
    @POST("api/v1/trainings")
    suspend fun create(@Body body: TrainingRecordCreateRequest): TrainingRecordResponse

    @GET("api/v1/trainings")
    suspend fun list(): List<TrainingRecordResponse>

    @GET("api/v1/trainings/statistics")
    suspend fun statistics(): TrainingStatisticsResponse
}
